src/components/SettingsPage/sections/Currencies.js

```diff
@@ -1,68 +1,208 @@
-        <TrackPage category="Settings" name="Currencies" currencyId={currency.id} />
-        <Header
-          icon={<IconCurrencies size={16} />}
-          title={t('settings.tabs.currencies')}
-          desc={t('settings.currencies.desc')}
-          renderRight={
-            <SelectCurrency
-              small
-              minWidth={200}
-              value={currency}
-              onChange={this.handleChangeCurrency}
-              currencies={currencies}
+      <Wrapper>
+        <TrackPage category="Settings" name="Currencies" />
+        <Section key={currency.id}>
+          <TrackPage category="Settings" name="Currencies" currencyId={currency.id} />
+          <Header
+            icon={<IconCurrencies size={16} />}
+            title={t('settings.tabs.currencies')}
+            desc={t('settings.currencies.desc')}
+            renderRight={
+              <Show visible={sectionVisible} onClick={this.toggleCurrencySection}>
+                <IconAngleDown size={24} />
+              </Show>
+            }
+          />
+          {sectionVisible && (
+            <Body>
+              <Row desc={t('settings.currencies.select')}>
+                <SelectCurrency
+                  small
+                  minWidth={200}
+                  value={currency}
+                  onChange={this.handleChangeCurrency}
+                  currencies={currencies}
+                />
+              </Row>
+              <CurrencyRows currency={currency} />
+            </Body>
+          )}
+        </Section>
+        <Section key={'rates'}>
+          <Header
+            icon={<IconActivity size={16} />}
+            title={t('settings.tabs.rates')}
+            desc={t('settings.rates.desc')}
+          />
+          <Body>
+            <RateRowWrapper>
+              <Box ff="Open Sans|SemiBold" alignItems="center" horizontal color="dark" fontSize={4}>
+                {'Rate'}
+                <TooltipButtonWrapper>
+                  <Tooltip render={RateTooltip}>
+                    <IconInfoCircle size={12} />
+                  </Tooltip>
+                </TooltipButtonWrapper>
+              </Box>
+              <Box ff="Open Sans|SemiBold" color="dark" fontSize={4}>
+                {'Price'}
+              </Box>
+              <Box ff="Open Sans|SemiBold" color="dark" fontSize={4}>
+                {'Last 30 days'}
+              </Box>
+              <Box ff="Open Sans|SemiBold" color="dark" fontSize={4}>
+                {'Exchange'}
+              </Box>
+            </RateRowWrapper>
+            <RateRow
+              from={intermediaryCurrency}
+              to={counterValueCurrency}
+              exchangeId={counterValueExchange}
             />
-          }
-        />
-        <Body>
-          <CurrencyRows currency={currency} />
-        </Body>
-      </Section>
+            {currencies.filter(c => c !== intermediaryCurrency).map(from => {
+              const cs = currenciesSettings[from.id]
```

Comment:
i'll fix it but we should not directly use `currenciesSettings[]` there are selector for that, and in fact that's where the bug come from: it's not from.id but it's from.ticker (today!)